### PR TITLE
[11.x] npm check in composer run dev command

### PR DESCRIPTION
**Pull request by amdad121**
I installed a fresh Laravel app and tried to run the composer run dev command, but it showed an error saying 'vite not found,' indicating that npm is not installed. I have added a check to see if npm is installed, and if not, it will automatically install npm.

laravel

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
         ],
         "dev": [
             "Composer\\Config::disableProcessTimeout",
+            "[ -f package-lock.json ] || npm install",
```
Comment (crynobone) [inline composer.json:51]: Should it better to use `npm ci` here?

Comment (amdad121) [reply composer.json:51]: Yes, I think it would be better.

**Comment by taylorotwell** [comment]
Does this file check work on Windows?

**Comment by PerryvanderMeer** [comment]
@taylorotwell Nope.

afbeelding

**Comment by amdad121** [comment]
> Does this file check work on Windows?

Unfortunately, the old code wasn't working on the Windows machine, so I changed the logic. I've now tested it on both my macOS and Windows machines, so it should work on Linux as well.

Screenshot 2024-10-16 212332
Screenshot 2024-10-16 212203
Screenshot 2024-10-16 at 9 37 44 PM
Screenshot 2024-10-16 at 9 38 56 PM

**Comment by amdad121** [comment]
@taylorotwell It's working fine in my machine as i said. Please check out the attached images.

**Pull request closed without merging**